tasks/sonar_config.yml

```diff
@@ -0,0 +1,69 @@
+---
+
+- name: 'create sonar database'
+  postgresql_db: name="{{ sonar_postgres_db_name }}" template='template0'
+                 encoding='UTF-8'
+                 #lc_collate='en_EN.UTF-8' lc_ctype='en_EN.UTF-8'
+  tags:
+    - sonar
+    - sonarconfig
+    - database
+    - postgresql
+
+- name: 'create sonar user'
+  postgresql_user: db=sonar name='{{ sonar_username }}' password='{{ sonar_password }}'
+  tags:
+    - sonar
+    - sonarconfig
+    - database
+    - postgresql
+
+# In order to add section to the sonar.properties file (thanks to assemble directive)
+- name: 'create sonar properties directory'
+  file:
+    dest: "{{ sonar_directory }}/conf/properties"
+    state: directory
+
+- name: 'configure sonar.properties'
+  template:
+    src: ../templates/sonar.properties
+    dest: "{{ sonar_directory }}/conf/properties/sonar.properties"
+  notify: restart sonar
+  tags:
+    - sonar
+    - sonarconfig
+
+- name: 'add directory for sonar-secret file'
+  file:
+    dest: "{{ sonar_home }}/.sonar"
+    mode: 0644
+    state: directory
+    owner: "{{ sonar_username }}"
+    group: "{{ sonar_username }}"
+
+- name: 'add sonar-secret file'
+  template:
+    src: ../templates/sonar-secret.txt
```

Comment:
I don't see a `sonar-secrets.txt` file in the repo? If you're just writing a secret to a file, I think you might want something like 
```yaml
- name: add sonar-secret file
  copy:
    content: "{{ sonar_secret }}"
    dest: "{{ sonar_home }}/.sonar"
    mode: 0644
    owner: "{{ sonar_username }}"
    group: "{{ sonar_username }}"
```

where `sonar_secret` is a variable with a sane default that is overwritten by a vaulted secret when the role is used

Reply:
Actually I looked into this, it seems that if you use variable interpolation as the content of the file, its recommended to use `template` instead

> If you need variable interpolation in copied files, use the template module. Using a variable in the content field will result in unpredictable output.

https://docs.ansible.com/ansible/latest/modules/copy_module.html?highlight=copy